### PR TITLE
Bumps current version to 8.12

### PR DESCRIPTION
**Pull request by abdonpijpelink**
This PR changes the "current" version of the documentation to 8.12.

Do not merge until release day.

Related to https://github.com/elastic/dev/issues/2441

### CODE REVIEW

**Comment by abdonpijpelink** [comment]
@kilfoyle Can you help out with these two docs sets?

"Starting with the Elasticsearch Platform and its Solutions" still has ` live:       [ 8.10 ]`. Should that be bumped to 8.12 on release day? If so, how about changing this into `live:       *stackcurrent`?

```
          - title:      "Starting with the Elasticsearch Platform and its Solutions"
            prefix:     en/starting-with-the-elasticsearch-platform-and-its-solutions
            current:    *stackcurrent
            index:      welcome-to-elastic/index.asciidoc
            branches:   [ {main: master}, 8.12, 8.11, 8.10, 8.9, 8.8, 8.7, 8.6, 8.5, 8.4, 7.17 ]
            live:       [ 8.10 ]
```

Should "Ingest: Add your data" be bumped to 8.12 on release day too? Or is it on its own release schedule? If the former, how about using `live:       *stackcurrent` here?

```
    -   title:     "Ingest: Add your data"
        sections:
          - title:      Elastic Ingest Reference Architectures
            prefix:     en/ingest
            current:    *stackcurrent
            index:      docs/en/ingest-arch/index.asciidoc
            branches:   [ {main: master}, 8.12, 8.11, 8.10, 8.9 ]
            live:       [ 8.11 ]
```

@leemthompo "Enterprise Search Node.js client" still has `live:       [ 8.10 ]`. Should that be bumped to 8.12 on release day? If so, how about changing this into `live:       *stackcurrent`?

```
              - title:      Enterprise Search Node.js client
                prefix:     enterprise-search-node
                current:    *stackcurrent
                branches:   [ {main: master}, 8.12, 8.11, 8.10, 8.9, 8.8, 8.7, 8.6 ]
                live:       [ 8.10 ]
```

**Comment by leemthompo** [comment]
I don't see any reason why live shouldn't be `*stackcurrent` for Enterprise Search Node.js client. @chriscressman might know more about this?

**Review by kilfoyle — DISMISSED** [review]
LGTM!

**Comment by kilfoyle** [comment]
> "Starting with the Elasticsearch Platform and its Solutions" still has  live:       [ 8.10 ]. Should that be bumped to 8.12 on release day? If so, how about changing this into live:       *stackcurrent?

I think `live:       *stackcurrent` should be fine here.

> Should "Ingest: Add your data" be bumped to 8.12 on release day too? Or is it on its own release schedule? If the former, how about using live:       *stackcurrent here?

And here as well. CC @karenzone since this is the "Elastic Ingest Reference Architectures".

**Comment by chriscressman** [comment]
> I don't see any reason why live shouldn't be *stackcurrent for Enterprise Search Node.js client.

Same. I think we should just change that one to `*stackcurrent`.

**Comment by karenzone** [comment]
> I think this also should be at 8.12 and could use the  live:       *stackcurrent.
CC @karenzone since this is the "Elastic Ingest Reference Architectures" guide.

Yes, @kilfoyle, that is correct. Thanks for asking.

**Comment by abdonpijpelink** [comment]
Thanks @karenzone @kilfoyle @chriscressman @leemthompo ! In https://github.com/elastic/docs/pull/2862/commits/83410d338a957c9a88fc7e8c52a0659b8bceff67, I've switched the following books to use `live: *stackcurrent` instead of a specific version:

- Starting with the Elasticsearch Platform and its Solutions
- Ingest: Add your data
- Enterprise Search Node.js client

**Review by leemthompo — APPROVED** [review]

**Comment by abdonpijpelink** [comment]
```
12:00:53 INFO:build_docs:Bad cross-document links:
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/java/1.x/config-core.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-process.html#max_event_size
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/java/1.x/ssl-configuration.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/agent-tls.html#agent-client-cert
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/java/current/config-core.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-process.html#max_event_size
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/java/current/ssl-configuration.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/agent-tls.html#agent-client-cert
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/java/master/config-core.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-process.html#max_event_size
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/java/master/ssl-configuration.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/agent-tls.html#agent-client-cert
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/nodejs/3.x/agent-api.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/api-metadata.html#api-metadata-schema
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/nodejs/3.x/configuration.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-process.html#max_event_size
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/nodejs/3.x/logs.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/log-correlation.html#ingest-logs-in-es
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/nodejs/3.x/release-notes-3.x.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-process.html#max_event_size
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/nodejs/4.x/agent-api.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/api-metadata.html#api-metadata-schema
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/nodejs/4.x/configuration.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-process.html#max_event_size
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/nodejs/4.x/logs.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/log-correlation.html#ingest-logs-in-es
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/nodejs/4.x/release-notes-3.x.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-process.html#max_event_size
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/nodejs/current/agent-api.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/api-metadata.html#api-metadata-schema
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/nodejs/current/configuration.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-process.html#max_event_size
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/nodejs/current/logs.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/log-correlation.html#ingest-logs-in-es
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/nodejs/current/release-notes-3.x.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-process.html#max_event_size
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/nodejs/master/agent-api.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/api-metadata.html#api-metadata-schema
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/nodejs/master/configuration.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-process.html#max_event_size
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/nodejs/master/logs.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/log-correlation.html#ingest-logs-in-es
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/nodejs/master/release-notes-3.x.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-process.html#max_event_size
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/rum-js/4.x/configuring-cors.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-rum.html#rum-allow-origins
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/rum-js/5.x/configuring-cors.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-rum.html#rum-allow-origins
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/rum-js/current/configuring-cors.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-rum.html#rum-allow-origins
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/rum-js/master/configuring-cors.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-rum.html#rum-allow-origins
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/agent/rum-js/master/sourcemap.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/source-map-how-to.html#source-map-rum-generate
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/source-map-how-to.html#source-map-rum-upload
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/cloud-heroku/current/ech-manage-apm-settings.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-rum.html#rum-library-pattern
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/cloud-heroku/latest/ech-manage-apm-settings.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-rum.html#rum-library-pattern
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/cloud/current/ec-manage-apm-settings.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-rum.html#rum-library-pattern
12:00:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/cloud/latest/ec-manage-apm-settings.html contains broken links to:
12:00:53 INFO:build_docs:   - en/apm/guide/8.12/configuration-rum.html#rum-library-pattern
```

**Comment by kilfoyle** [comment]
I'll fix the four Cloud broken links.

**Comment by abdonpijpelink** [comment]
@elasticmachine run elasticsearch-ci/docs

File: conf.yaml [hunk 1]
```diff
@@ -1430,7 +1430,7 @@ contents:
             current:    *stackcurrent
             index:      docs/en/ingest-arch/index.asciidoc
             branches:   [ {main: master}, 8.12, 8.11, 8.10, 8.9 ]
-            live:       [ 8.11 ]
+            live:       [ *stackcurrent ]
```
Comment (karenzone) [inline conf.yaml:1433]: Thanks for catching and fixing for ingest architectures. <3

**Comment by abdonpijpelink** [comment]
buildkite test this rebuild

**Pull request merged**